src/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+file(GLOB files_decomp factor.f90
+	               decomp_2d_constants.f90
+	               decomp_2d.f90
+		       log.f90
+		       io.f90)
+include_directories(${CMAKE_SOURCE_DIR}/src)
+include_directories(${CMAKE_SOURCE_DIR}/mod)
+
+#if(${FFT_Choice} MATCHES "generic")
+#  file(GLOB files_fft fft_generic.f90)
+#endif(${FFT_Choice} MATCHES "generic")
+
+if(FFTW_FOUND)
+  message (STATUS "Compiling using FFTW3")
+  file(GLOB files_fft fft_fftw3.f90)
+elseif(MKL_FOUND)
+  message (STATUS "Compiling using MKL")
+  file(GLOB files_fft $ENV{MKLROOT}/include/mkl_dfti.f90 fft_mkl.f90)
```

Comment:
Although commit [d90c665](https://github.com/xcompact3d/2decomp-fft/pull/142/commits/d90c6656953d2b07f355c49a28c180ba30d8ffee) enables linking MKL (apparently successfully), the resulting library does not pass all tests. Using the generic FFT with Intel compilers successfully passes tests.

Reply:
Passes when built in `DEV` and `DEBUG` modes